retrieverdash/dashboard_script/dashboard_script.py

```diff
@@ -1,35 +1,38 @@
 import json
 import os
-from multiprocessing import Pool
+from datetime import datetime, timezone
+from json.decoder import JSONDecodeError
 from shutil import rmtree
 from tempfile import mkdtemp
 
-from filelock import FileLock
 from retriever import datasets
 from retriever import download
 from retriever.lib.engine_tools import getmd5
 
-from status_dashboard_tools import get_dataset_md5
-from status_dashboard_tools import diff_generator
-from status_dashboard_tools import create_dirs
-from status_dashboard_tools import dataset_type
+from .status_dashboard_tools import get_dataset_md5
+from .status_dashboard_tools import diff_generator
+from .status_dashboard_tools import create_dirs
+from .status_dashboard_tools import dataset_type
 
-file_location = os.path.dirname(os.path.realpath(__file__))
+file_location = os.path.normpath(os.path.dirname(os.path.realpath(__file__)))
+
+IGNORE_LIST = ['prism-climate', 'mammal-super-tree', 'forest-inventory-analysis',
+               'vertnet']
 
 
 def check_dataset(dataset):
-    os.chdir(os.path.join(file_location))
     md5 = None
     status = None
     reason = None
     diff = None
+    dataset_detail = None
     try:
         try:
-            dataset_detail = json.load(open('dataset_details.json', 'r'))
-        except FileNotFoundError:
-            with open("dataset_details.json", 'w') as json_file:
-                dataset_detail = dict()
-                json.dump(dataset_detail, json_file)
+            with open("dataset_details.json", 'r') as json_file:
+                dataset_detail = json.load(json_file)
+        except (OSError, JSONDecodeError):
+            dataset_detail = dict()
+            dataset_detail['dataset_details'] = {}
```

Comment:
```python
     except Exception as e:
        reason = str(e)	        reason = str(e)
        status = False	        status = False
```
Do we want to type cast this to str()? pyth3 str() is not the same as python2 str()

Reply:
Yes, I did this it to write to json file. Is there any issue?